src/components/Atoms/Input/index.jsx

```diff
@@ -1,16 +1,45 @@
-import React from "react";
+import React, { useRef, useState, useEffect } from "react";
 import styled from "styled-components";
 import COLOR from "../../../variables/color";
-import pencil from "../../../assets/svg/pencil.svg";
+import TEXTS from "../../../variables/texts";
+
+<InputTask defaultValue onEditComplete /> 
 
 const InputTask = (props) => {
-  return <InputFrame onClick = {props.onClick}>
-    <img src={pencil} width={"20px"} />
+  const inputRef = useRef(null);
+
+  const focusout = () => {
+    props.onEditComplete(inputRef.current.value);
+  };
+
+  const keypress = (e) => {
+    if(e.code===`Enter`)
+    {
+      inputRef.current.removeEventListener('blur',focusout,false);
+      props.onEditComplete(inputRef.current.value);
+    }
+  }
+
+  useEffect(() => {
+    inputRef.current.value = props.defaultValue;
+    inputRef.current.focus();
+    inputRef.current.addEventListener('blur',focusout,false);
+    inputRef.current.addEventListener('keydown',keypress,false);
+  }, [])
+
+  return <InputFrame ref={inputRef} type="text">
+    
   </InputFrame>;
```

Comment:
```suggestion
  return <InputFrame ref={inputRef} type="text"/>;
```
こう書くこともできます。中身無い場合はコンパクトなのでこっちでよいかも